file.go

```diff
@@ -653,3 +653,61 @@ func (f *File) createFileADV() error {
 
 	return nil
 }
+
+// CalculateBalancedOffset will append a "balanced offset" record to the end of an ACH File.
+// Balanced offset records sum a File's Debits or Credits to zero in a single transaction.
+// Offset files are used to offset transactions from a single account inside of the ODFI.
+//
+// If a file contains Debits the offset record will be a Credit and if a file contains Credits
+// the offset record will be a debit.
+//
+// The ACH File will have its FileControl record retabulated after appending the Batch.
+func (f *File) CalculateBalancedOffset(bh *BatchHeader, ed *EntryDetail) error {
+	if err := f.Create(); err != nil {
+		return fmt.Errorf("CalculateBalancedOffset: initial tabulate: %v", err)
+	}
+	if f.Control.TotalDebitEntryDollarAmountInFile > 0 && f.Control.TotalCreditEntryDollarAmountInFile > 0 {
```

Comment:
Should that be BatchControl instead of FileControl?

Reply:
I thought offset records were for an entire file.

Reply:
My understanding was offset records are used in batches, to create a balanced file.

Reply:
The current library creates ach files which contain batches with detail entries the caller specifies.  The caller can create an offset entryDetail in the batch.  My understanding of this enhancement is to add a helper function to create an offset entry so the caller doesn't have to create the offset entry.